### ISSUE TITLE
`middle(::TimeSpan)`

### ISSUE THREAD

**Issue by palday** [beacon-biosignals/TimeSpans.jl#24]
I think it might make sense to define

```julia
Statistics.middle(t::TimeSpan) = (start(t) + stop(t)) / 2
```
I have a few uses for it. If it sounds generally useful, I'll open a PR.

**Comment by ararslan** [comment]
FWIW I'm not sure we'd be able to use that definition verbatim, since `start(t) + stop(t)` will be a `Nanosecond` and may be odd, so dividing by 2 will throw an `InexactError`.

Relatedly, what should `middle(TimeSpan(0, 1))` return? It can't actually be represented in the precision provided by TimeSpans, nor by the Dates stdlib.

**Comment by palday** [comment]
Yeah, the edge cases occurred to me later. In both cases, I would say round down to the nearest whole Nanosecond because the spans are half-open intervals, so rounding down is line with "include the lower bound but not the upper".

**Comment by ararslan** [comment]
So I guess the definition would be `fld(start(t) + stop(t), 2)` then.
